Library/Homebrew/cmd/install.rb

```diff
@@ -98,10 +98,27 @@ def install
       else
         ofail e.message
         query = query_regexp(e.name)
-        ohai "Searching formulae..."
-        puts_columns(search_formulae(query))
+
+        ohai "Searching for similarly named formulae..."
+        formulae_search_results = search_formulae(query)
+
+        if !formulae_search_results.any?
+          ofail 'No similarly named formulae found.'
+        else
+          puts 'These similarly named formulae were found:'
+          puts_columns(formulae_search_results)
+          puts "If you meant to install one of them, run `brew install <formula>`"
```

Comment:
We could make a special case for when there’s only one result, e.g. “``Run `brew install <formula>` to install it``”.